lib/Parse/ParseDecl.cpp

```diff
@@ -5616,33 +5664,61 @@ bool swift::isKeywordPossibleDeclStart(const LangOptions &options,
 }
 
 static bool
-isParenthesizedModifier(Parser &P, StringRef name,
-                        std::initializer_list<StringRef> allowedArguments) {
+consumeIfParenthesizedModifier(Parser &P, StringRef name,
+                               std::initializer_list<StringRef> allowedArguments) {
   assert((P.Tok.getText() == name) && P.peekToken().is(tok::l_paren) &&
          "Invariant violated");
 
   // Look ahead to parse the parenthesized expression.
-  Parser::BacktrackingScope Backtrack(P);
+  Parser::CancellableBacktrackingScope backtrack(P);
   P.consumeToken(tok::identifier);
   P.consumeToken(tok::l_paren);
+  
+  const Token &Tok2 = P.peekToken();
+  if (P.consumeIf(tok::code_complete)) {
+    // If a code complete token is present, recover from missing/incorrect argument and missing '('
+    P.consumeIf(tok::identifier);
+    P.consumeIf(tok::r_paren);
+    backtrack.cancelBacktrack();
+    return true;
+  }
+  
+  if (P.Tok.is(tok::identifier) && !P.Tok.isContextualDeclKeyword() && Tok2.is(tok::code_complete)) {
+    P.consumeToken(tok::identifier);
+    P.consumeToken(tok::code_complete);
+    
+    // If a code complete is present with a non-keyword token before it, recover from missing/incorrect argument and missing '('
+    if (!P.Tok.isContextualDeclKeyword()) {
+      P.consumeIf(tok::identifier);
+    }
+    P.consumeIf(tok::r_paren);
+    backtrack.cancelBacktrack();
+    return true;
+  }
```

Comment:
Similar to https://github.com/swiftlang/swift/pull/79974#discussion_r1994321852, I think this isn’t actually needed. Or am I missing something?

Reply:
I've removed it. 👍🏼